spiketoolkit/sortingcomponents/detection.py

```diff
@@ -116,11 +112,17 @@ def detect_spikes(recording, channel_ids=None, detect_threshold=5, n_pad_ms=2, u
     all_channel_times = [[] for ii in range(len(channel_ids))]
     all_channel_amps = [[] for ii in range(len(channel_ids))]
 
+    snippet_len = int(snippet_size_sec * recording.get_sampling_frequency())
+    reference_frames = np.linspace(snippet_len+1, recording.get_num_frames() - snippet_len,
+                                   n_snippets_for_threshold)
+    snippets = recording.get_snippets(reference_frames=reference_frames, snippet_len=snippet_len)
+    traces_mad = np.concatenate(snippets, 1)
+    thresholds = detect_threshold * np.median(np.abs(traces_mad) / 0.6745, 1)[:, None]
+
     if n_jobs > 1:
         output = Parallel(n_jobs=n_jobs)(delayed(_detect_and_align_peaks_chunk)
-                                         (ii, rec_arg, chunks, channel_ids, detect_threshold,
-                                                              detect_sign, n_pad_samples, upsample, min_diff_samples,
-                                                              align, verbose)
+                                         (ii, rec_arg, chunks, channel_ids, thresholds, detect_sign,
+                                          n_shifts, verbose)
```

Comment:
What is annoying here is that "thresholds" is serialize at each loop. Maybe this have a little cost.
maybe we could memmap it.